src/main/java/hudson/plugins/sonar/utils/SonarUtils.java

```diff
@@ -88,13 +88,25 @@ public static String extractSonarProjectURLFromLogs(AbstractBuild<?, ?> build) t
    * Iterate previous build of this project and return the last Sonar URL
    */
   public static String getLastSonarUrl(AbstractProject<?, ?> project) {
-    RunList<? extends Run<?, ?>> builds = project.getBuilds();
-    for (Run<?, ?> run : builds) {
-      BuildSonarAction action = run.getAction(BuildSonarAction.class);
-      if (action != null) {
-        return action.getUrlName();
-      }
+    String url = getSonarUrlFromRun(project.getLastSuccessfulBuild());
+    if (url != null) {
+      return url;
+    } else {
+      return getSonarUrlFromRun(project.getLastUnstableBuild());
+    }
+  }
+
+  @Nullable
```

Comment:
Use `@CheckForNull`. In general avoid `@Nullable` unless you have read its Javadoc carefully and really know what you are doing.